unit_tests/test_tlslite_utils_tripledes.py

```diff
@@ -0,0 +1,533 @@
+                             b'\x55\xfe\x07\x2a\x73\x51\xa5\x00')
+
+    @unittest.skipIf(PY_VER < (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test1_py3_str_instance(self):
+
+        with self.assertWarns(DeprecationWarning):
+            Python_TripleDES('asdfdasdfdsasdfdsasgdfds', b"\xbb"*8)
+
+    @unittest.skipIf(PY_VER < (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test2_py3_str_instance(self):
+
+        with self.assertWarns(DeprecationWarning):
+            Python_TripleDES(b"\xaa"*24, 'asdfdasd')
+
+    @unittest.skipIf(PY_VER < (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test3_py3_str_instance(self):
+        key =  b"\xaa"*24
+        iv = b"\xbb"*8
+
+        with self.assertWarns(DeprecationWarning):
+            Python_TripleDES(key, iv).encrypt('161514131211109876543210')
+
+    @unittest.skipIf(PY_VER < (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test4_py3_str_instance(self):
+        key =  b"\xaa"*24
+        iv = b"\xbb"*8
+
+        with self.assertWarns(DeprecationWarning):
+            Python_TripleDES(key, iv).decrypt('161514131211109876543210')
+
+    @unittest.skipIf(PY_VER > (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test1_py2_str_instance(self):
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            Python_TripleDES('asdfdasdfdsasdfdsasgdfds', b"\xbb"*8)
+            self.assertEqual(len(w), 0)
+
+    @unittest.skipIf(PY_VER > (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test2_py2_str_instance(self):
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            Python_TripleDES(b"\xaa"*24, 'asdfdasd')
+            self.assertEqual(len(w), 0)
+
+    @unittest.skipIf(PY_VER > (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test3_py2_str_instance(self):
+        key =  b"\xaa"*24
+        iv = b"\xbb"*8
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            Python_TripleDES(key, iv).encrypt('161514131211109876543210')
+            self.assertEqual(len(w), 0)
+
+    @unittest.skipIf(PY_VER > (3, ),
+        "DeprecationWarning check should apply only on python3")
+    def test4_py2_str_instance(self):
+        key =  b"\xaa"*24
+        iv = b"\xbb"*8
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            Python_TripleDES(key, iv).decrypt('161514131211109876543210')
+            self.assertEqual(len(w), 0)
+
+    @unittest.skipIf(PY_VER < (3, ), "Unicode string syntax on py2: u'ľáľä'")
+    def test1_py3_unicode_instance(self):
+        with self.assertRaises(ValueError):
```

Comment:
this should also assert a warning (so two `with`, one inside the other) otherwise it pollutes the output of unittest run